api/server/runner.go

```diff
@@ -37,7 +37,7 @@ func (s *Server) handleFunctionCall2(c *gin.Context) error {
 	}
 
 	appID := c.MustGet(api.AppID).(string)
-	app, err := s.datastore.GetAppByID(c.Request.Context(), appID)
+	app, err := s.agent.GetAppByID(c.Request.Context(), appID)
```

Comment:
this should prob get pushed down into a `GetCall` option, `agent` shouldn't expose GetAppByID, but we can wait and I'll add it to my ever expanding list of todos